google/cloud/security/common/gcp_type/instance.py

```diff
@@ -136,3 +145,71 @@ def name(self):
             str: name
         """
         return self._path_component('name')
+
+
+# pylint: disable=too-few-public-methods
+class InstanceNetworkInterface(object):
+    """InstanceNetworkInterface Resource."""
+
+    def __init__(self, **kwargs):
+        """Initialize
+
+        Args:
+            kwargs: json from a single instance on the network_interfaces
+        """
+        self.kind = kwargs.get('kind')
+        self.network = kwargs.get('network')
+        self.subnetwork = kwargs.get('subnetwork')
+        self.network_ip = kwargs.get('networkIP')
+        self.name = kwargs.get('name')
+        self.access_configs = kwargs.get('accessConfigs')
+        self.alias_ip_ranges = kwargs.get('aliasIpRanges')
+
+    def __repr__(self):
+        """Repr
+
+        Returns:
+            string: a string for a InstanceNetworkInterface
+        """
+        return 'kind: %s Network: %s subnetwork: %s network_ip %s name %s' \
+            'access_configs %s alias_ip_ranges %s' % (
+                self.kind, self.network, self.subnetwork, self.network_ip,
+                self.name, self.access_configs, self.alias_ip_ranges)
```

Comment:
The value of access_configs is a dict currently, so it's printed order is non-deterministic. Two identical networks could have different __repr__ strings because of that, and fail the __hash__ check. You could flatten access_configs into a sorted tuple, which is done in other places in the forseti code base, like common/gcp_type/key.py.

Another option would be to export the dict to a json string with sorted keys and store that in a private attribute along with the sha1 hash of the json string. You can then use use the hash bytes for the value of __hash__ and for comparison in __eq__ and __ne__.